### PR TITLE
feat: add flag to change working directory

### PR DESCRIPTION
**Pull request by Fubukimaru**
I have added a new `-r` flag to change the working directory of the assembler to the path where the .asm file is.

Tests and documentation have been updated to cover the Issue #133.

### CODE REVIEW

File: src/dura.y [hunk 1]
```diff
@@ -4622,14 +4626,35 @@ int main(int argc, char *argv[]) {
...
 
-  strncpy(fname_no_ext, argv[fileArg], PATH_MAX);
-  strncpy(fname_asm, fname_no_ext, PATH_MAX);
+  // Getting last file as src file (using fileArg, counter of files)
+  strncpy(fname_asm, argv[fileArg], PATH_MAX);
+
+  if (relative_path) {
+    // Reusing array to change dir, latter it will have the correct values
+    strncpy(fname_no_ext, fname_asm, PATH_MAX);
+    for (t = strlen(fname_no_ext) - 1; (fname_no_ext[t] != '/') && t; t--);
```
Comment (duhow) [inline src/dura.y:4635]: esto no funcionará en Windows...

**Review by duhow — CHANGES_REQUESTED** [review]

File: src/dura.y (hunk 1 above)
Comment (Fubukimaru) [reply src/dura.y]: https://github.com/Fubukimaru/asMSX/pull/134/commits/b6168e4252820605931259ed96c7a3cb032fb3b3 Will this work?

**Review by duhow — DISMISSED** [review]

File: test/steps/build.py [hunk 2]
```diff
@@ -13,10 +13,22 @@ def step_impl(context, folder):
...
 @step('file {file} exists')
-def step_impl(context, file):
-    assert os.path.isfile(file), f"File {file} does not exist"
+@step('file {file} does {should} exist')
+def step_impl(context, file, should: str = ""):
+    should = (not "no" in should)
+    if should:
+        assert os.path.isfile(file), f"File {file} does not exist"
+    else:
+        assert not os.path.isfile(file), f"File {file} does exist"
```
Comment (ProtossGP32) [inline test/steps/build.py:31]: @Fubukimaru Correct me if I'm wrong, but aren't the messages in the `assert` defined backwards?
- `if should` --> The file should exists, then the assert returns `f"File {file} does exist"`
- `else` --> The file shouldn't exist, then the assert returns `f"File {file} does exist"`

Comment (duhow) [reply test/steps/build.py:31]: if the file `should` exist, check it does exist, otherwise throw `file does not exist`

Comment (ProtossGP32) [reply test/steps/build.py:31]: Thanks! I misread the code and thought that it was the message that the assert returned, not the exception

**Comment by Fubukimaru** [comment]
@jamque Can you check again if it works on Windowzer?
https://github.com/Fubukimaru/asMSX/actions/runs/11643932198/artifacts/2136513316

**Comment by Fubukimaru** [comment]
Got the Ok from @jamque.

@duhow You can proceed with the review.

Muito obrigado 🙇

**Review by duhow — APPROVED** [review]

**Comment by Fubukimaru** [comment]
💃💃💃💃💃💃💃💃💃💃💃 YAS 💃💃💃💃💃💃💃💃💃💃💃

**Pull request merged**